src/pytorch_lightning/core/module.py

```diff
@@ -424,6 +424,7 @@ def log(
             )
 
         value = apply_to_collection(value, numbers.Number, self.__to_tensor)
+        value = apply_to_collection(value, torch.Tensor, torch.squeeze)
```

Comment:
Why does it not get moved to device if it's not squeezed?

Reply:
the issue is
```py
torch.lt(torch.Tensor([1.2], device=cuda), torch.Tensor(1.2, device=cpu)) -> gives an error
torch.lt(torch.Tensor(1.2), device=cuda), torch.Tensor(1.2, device=cpu)) -> works
```
with logging, we anyways deal with scalars.
we don't need them on the same device.